nixos/modules/services/monitoring/alertmanager-executor.nix

```diff
@@ -0,0 +1,70 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.alertmanager-executor;
+  settingsFormat = pkgs.formats.yaml {};
+
+in {
+  options = {
+    services.alertmanager-executor = {
+      enable = mkEnableOption "alertmanager-executor (aka prometheus-am-executor)";
+
+      user = mkOption {
+        type = types.str;
+        default = "alertmanager-executor";
+        description = lib.mdDoc "User to run the service and executed commands under.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "alertmanager-executor";
+        description = lib.mdDoc "Group to run the service and executed commands under.";
+      };
+
+      settings = mkOption {
+        type = settingsFormat.type;
+        default = {};
+        example = {
+          listen_address = "localhost:23222";
+          commands = [
+            {
+              cmd = "echo";
+              args = [ "alert" "firing" ];
+            }
+          ];
+        };
+        description = lib.mdDoc ''
+          Configuration for alertmanager-executor. Documentation:
+          <https://github.com/imgix/prometheus-am-executor#configuration-file-format>.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.users = optionalAttrs (cfg.user == "alertmanager-executor") {
+      alertmanager-executor = {
+        description = "Execute command based on Prometheus alerts";
+        isSystemUser = true;
+        group = cfg.group;
+      };
+    };
+
+    users.groups = optionalAttrs (cfg.group == "alertmanager-executor") {
+      alertmanager-executor = {};
+    };
```

Comment:
Is it a requirement to have a separate user here? Can we just use `DynamicUser`? If no, we should add this user [here](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/misc/ids.nix) first.

Reply:
My use-case needs the service to run under particular user, I suppose this requirement is not unusual since alertmanager-executor seems well suited for integrating alertmanager with 3rd party services using command-line tools. Not sure about ids.nix, looking at the file it seems that static ids should only be used when really necessary.

Reply:
Hmm, I still think we should add to ids since the service could write and read from multiple files, let's see what other people think about this.

Reply:
[Picked a random PR to try to progress]

I think this is the only outstanding review comment here, and per https://github.com/NixOS/rfcs/blob/master/rfcs/0052-dynamic-ids.md / https://github.com/NixOS/nixpkgs/commit/4971ed01cbc149465443c5107ba3331d8fda39cc it looks like best practice is specifically to not specify static IDs.